core/java/com/google/instrumentation/stats/StatsContext.java

```diff
@@ -51,12 +51,13 @@ public final StatsContext with(
   /**
    * Serializes the {@link StatsContext} into the on-the-wire representation.
    *
-   * <p>The inverse of {@link StatsContextFactory#deserialize(ByteBuffer)} and should be based on
+   * <p>The inverse of {@link StatsContextFactory#deserialize(OutputStream)} and should be based on
    * the {@link StatsContext} protobuf representation.
    *
-   * @return serialized bytes.
+   * @param output the {@link OutputStream} to add the serialized form of this {@link StatsContext}.
+   * @return the given {@link OutputStream}
    */
-  public abstract ByteBuffer serialize();
+  public abstract OutputStream serialize(OutputStream output);
```

Comment:
I have it return the output stream in case someone wants to chain calls together.

I am thinking of adding a 'throws IOException' in case writing to the stream has a problem.